src/libcore/num/dec2flt/parse.rs

```diff
@@ -58,25 +58,23 @@ pub enum ParseResult<'a> {
 pub fn parse_decimal(s: &str) -> ParseResult {
     let s = s.as_bytes();
     let (integral, s) = eat_digits(s);
+
     match s.first() {
-        None => {
-            if integral.is_empty() {
-                return Invalid; // No digits at all
-            }
-            Valid(Decimal::new(integral, b"", 0))
-        }
+        None if integral.is_empty() => Invalid, // No digits at all
+        None => Valid(Decimal::new(integral, b"", 0)),
```

Comment:
I hastily added this branch in #29050 but I now believe it would be better to check for empty input at the start of the function. Since you're touching this condition anyway, I'd like you to move it there.